File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.java
class Solution {
    public double minimumAverage(int[] nums) {
        double minEl = Integer.MAX_VALUE;
        Arrays.sort(nums);
        int i = 0, j = nums.length - 1;
        while (i <= j) {
            double avg = (double) (nums[i] + nums[j]) / 2;
            minEl = Math.min(minEl, avg);
            i++;
            j--;
        }
        return minEl;
    }
}

File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        minEl = float('inf')
        nums.sort()
        i, j = 0, len(nums) - 1
        while i <= j:
            avg = (nums[i] + nums[j]) / 2
            minEl = min(minEl, avg)
            i += 1
            j -= 1
        return minEl